homeassistant/components/vicare/config_flow.py

```diff
@@ -0,0 +1,110 @@
+"""Config flow for ViCare integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from PyViCare.PyViCareUtils import PyViCareInvalidCredentialsError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.dhcp import MAC_ADDRESS
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+
+from . import vicare_login
+from .const import (
+    CONF_CIRCUIT,
+    CONF_HEATING_TYPE,
+    DEFAULT_HEATING_TYPE,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    HeatingType,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for ViCare."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input: dict[str, Any] | None = None):
+        """Invoke when a user initiates a flow via the user interface."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        data_schema = {
+            vol.Required(CONF_USERNAME): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Required(CONF_CLIENT_ID): cv.string,
+            vol.Required(CONF_HEATING_TYPE, default=DEFAULT_HEATING_TYPE.value): vol.In(
+                [e.value for e in HeatingType]
+            ),
+            vol.Optional(CONF_NAME, default="ViCare"): cv.string,
+            vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): vol.All(
+                vol.Coerce(int), vol.Range(min=30)
+            ),
+        }
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                await self.hass.async_add_executor_job(
+                    vicare_login, self.hass, user_input
+                )
+                return self.async_create_entry(
```

Comment:
Please only wrap the line that can raise in the try... except block. Use and `else:` block if needed.